### PR TITLE
Add category field to edit model configuration page

### PR DESCRIPTION
**Pull request by ohltyler**
*Issue #, if available:*

*Description of changes:*

This PR adds the category field component on the edit model configuration page. This is an optional field that, when selected, will create and run a high cardinality / multi-entity detector.

Changes include:
- Adds `categoryField` field to the detector interface
- Adds category field component to the edit model config page (including validation & edge case of no eligible fields)
- Refactors and adds the `categoryField` field to the Additional settings component on detector configuration page
- Adds unit tests

All wording has been approved by UX and tech writer.

Screenshots:

1. Category field (unselected):
Screen Shot 2020-10-12 at 7 53 57 PM

2. Category field (selected):
Screen Shot 2020-10-12 at 7 58 22 PM

3. Category field (no eligible fields in index):
Screen Shot 2020-10-12 at 7 55 31 PM

4. Category field under 'Additional settings':
Screen Shot 2020-10-12 at 7 55 05 PM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/pages/EditFeatures/components/CategoryField/CategoryField.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  getError,
+  validateCategoryField,
+} from '../../../../utils/utils';
+//@ts-ignore
+import chrome from 'ui/chrome';
+
+interface CategoryFieldProps {
+  isHCDetector: boolean;
+  categoryFieldOptions: string[];
+  setIsHCDetector(isHCDetector: boolean): void;
```
Comment (yizheliu-amazon) [inline public/pages/EditFeatures/components/CategoryField/CategoryField.tsx:48]: you may delete this method, I think original `CategoryFieldProps` is already good for integration

**Review by yizheliu-amazon — APPROVED** [review]
Looks good to me. Feel free to merge it after addressing my comment.

File: public/pages/EditFeatures/components/CategoryField/CategoryField.tsx (hunk 1 above)
Comment (ohltyler) [reply public/pages/EditFeatures/components/CategoryField/CategoryField.tsx:48]: I added this to update the `isHCDetector` flag in `editFeatures`. This was needed to properly validate the form in the case of the box being checked, but no value being set.

Comment (ohltyler) [reply public/pages/EditFeatures/components/CategoryField/CategoryField.tsx:48]: Basically, since we don't perform validation immediately after enabling the checkbox now, this became an issue that had to be fixed by updating `isHCDetector` in `editFeatures`, so we can properly validate/invalidate in `handleSaveChanges()`.

Comment (yizheliu-amazon) [reply public/pages/EditFeatures/components/CategoryField/CategoryField.tsx:48]: Got it. I thought you may delete it since I already modified code to support validation only when checkbox is enabled. But I think your new implementation is better. Approved.

**Comment by yizheliu-amazon** [comment]
When I open Edit feature page for non-HC detector in my local workspace, the `no eligible fields in index` callout will show up in the beginning and it will be gone very soon, does that happen to you as well?

**Comment by ohltyler** [comment]
> When I open Edit feature page for non-HC detector in my local workspace, the `no eligible fields in index` callout will show up in the beginning and it will be gone very soon, does that happen to you as well?

I see, yes it does flash that as it loads in the index details. Let me look into improving the loading state.

**Comment by ohltyler** [comment]
> > > When I open Edit feature page for non-HC detector in my local workspace, the `no eligible fields in index` callout will show up in the beginning and it will be gone very soon, does that happen to you as well?
> > 
> > 
> > I see, yes it does flash that as it loads in the index details. Let me look into improving the loading state.
> 
> I will see it as non-blocking issue. If it takes too long to fix it, let's create issue for it and improve it later.

really quick fix. added test case as well

**Review by saratvemulapalli — COMMENTED** [review]
Changes look to me.

**Review by saratvemulapalli — APPROVED** [review]
